src/callbacks_step/amr_dg1d.jl

```diff
@@ -284,14 +284,11 @@ function coarsen_elements!(u::AbstractArray{<:Any, 3}, element_id,
     end
 end
 
-# this method is called when an `ControllerThreeLevel` is constructed
-function create_cache(::Type{ControllerThreeLevel}, mesh::TreeMesh{1}, equations,
-                      dg::DG, cache)
-    controller_value = Vector{Int}(undef, nelements(dg, cache))
-    return (; controller_value)
-end
-
-function create_cache(::Type{ControllerThreeLevelCombined}, mesh::TreeMesh,
+# Construct cache for ControllerThreeLevel and ControllerThreeLevelCombined.
+# This method is called when a controller is constructed
+function create_cache(::Union{Type{ControllerThreeLevel},
```

Comment:
Would it make sense to put this method in amr_dg.jl instead since it is not dimension-dependent anymore?

Reply:
Or probably even in amr.jl?

Reply:
[Yes](https://github.com/trixi-framework/Trixi.jl/blob/1e1f64314d49386ef1a671c0cb5dfda13a048c31/src/callbacks_step/amr_dg.jl)

Reply:
@sloede Does moving this stuff violate your "better be explicit and duplicate code for several space dimensions" policy?

Reply:
I believe there is precedence where we have replaced literally-identical implementations with a single function (e.g., solver functions for the different mesh types). So yes, I'd be ok with that :-)

Reply:
I moved the `create_cache` routine into `amr_dg.jl` in 4be6415c06949658cab045e31e5d9ca3afdb8167.